hazelcast/src/main/java/com/hazelcast/spi/impl/SlowOperationLog.java

```diff
@@ -208,8 +211,10 @@ public JsonObject toJson() {
         public void fromJson(JsonObject json) {
             id = getInt(json, "id");
             startedAt = getLong(json, "startedAt");
-            durationNanos = getLong(json, "duration");
             lastAccessNanos = System.nanoTime();
+
+            // write durationMs last to enforce a write barrier
+            durationMs = getInt(json, "durationMs");
```

Comment:
I don't understand the reason of mentioning about a memory barrier here. Isn't this just a volatile field initialization? People already know about `volatile` semantics, I guess nothing more special done here...